packages/framework/src/Controller/Admin/CustomerController.php

```diff
@@ -296,6 +334,11 @@ public function deleteAction($id)
     public function loginAsUserAction($customerUserId)
     {
         $customerUser = $this->customerUserFacade->getCustomerUserById($customerUserId);
+
+        if ($this->domain->getId() !== $customerUser->getDomainId()) {
```

Comment:
I am thinking - this way you will never be able to log as a customer on the 2nd domain when accessing administration on the 1st domain, and vice versa. And moreover, the admin will not know what is happening (no flash message here). Or am I missing something